components/shared/ui/menu.tsx

```diff
@@ -1,53 +1,99 @@
 import Link from "next/link";
 import { FC, useState } from "react";
-import { mainColorBgClass } from "../../../lib/constants/colors";
+import { mainColorBgClass, mainColorBorderClass } from "../../../lib/constants/colors";
 import { useSiteCodename } from "../siteCodenameContext";
 import { Block_Navigation } from "../../../models";
 import { createItemSmartLink } from "../../../lib/utils/smartLinkUtils";
 import { ChevronDownIcon, Bars3Icon } from "@heroicons/react/24/solid";
+import { useRouter } from "next/router";
 
 type Link = Readonly<Block_Navigation>;
 
 type Props = Readonly<{
   item: Link;
 }>
 
+type MenuItemsProps = Readonly<{
+  items: Block_Navigation[];
+  activeMenu: string | number;
+  handleClick: (menuId: string | number) => void;
+}>
+
 type DropdownMenuProps = Readonly<{
   links: ReadonlyArray<Link>;
 }>;
 
-const DropdownMenu: FC<DropdownMenuProps> = props => {
+const MenuItems: FC<MenuItemsProps> = props => {
+  const siteCodename = useSiteCodename();
+  const router = useRouter();
+
   return (
     <>
-      {props.links.map(link => (
-        <Link
-          key={link.system.codename}
-          {...link.elements.openInANewWindow.value[0] ? { rel: "noopener noreferrer", target: "_blank" } : {}}
-          href={link.elements.externalLink.value ? link.elements.externalLink.value : "/" + link.elements.pageLink.linkedItems[0].elements.url.value}
-          className="block p-3 rounded-lg hover:bg-gray-50"
-        >
-          <div className="font-semibold">{link.elements.label.value}</div>
-          <span className="text-sm text-gray-500">{link.elements.caption.value}</span>
-        </Link>
+      {props.items.map((link, i) => (
+        <li key={i} className={`${"/" + link.elements.pageLink.linkedItems[0]?.elements.url.value === router.asPath ? "" : "border-l-transparent border-t-transparent"}
+        border-gray-500 border-l-8 border-t-0 md:border-t-8 md:border-l-0 h-full ${mainColorBgClass[siteCodename]} group grow`} onClick={() => props.handleClick(i)}>
+          {link.elements.subitems.value.length > 0 ? (
+            <div className={`${i === props.activeMenu ? "bg-white " : ""} md:hover:bg-white h-full`}>
+              <DropdownButton item={link} />
+              <div
+                className={`${i === props.activeMenu ? "block" : "hidden"} md:group-hover:block absolute z-50 left-0 shadow-sm bg-white border-gray-200 w-full`}
+              >
+                <div className="grid gap-2 max-w-screen-xl px-4 py-5 mx-auto text-gray-900 sm:grid-cols-2 md:grid-cols-3 md:px-6">
```

Comment:
for better semantics, I think this should be <ul>

Reply:
?

Reply:
rewritten to UL as the root component